### PR TITLE
SUS-1276 | refactor search code for an old corporate wiki

### PR DESCRIPTION
**Pull request by macbre**
https://wikia-inc.atlassian.net/browse/SUS-1276

Do not rely on `wgEnableWikiaHomePageExt` WikiFactory variable.

### CODE REVIEW

File: includes/wikia/Wikia.php [hunk 1]
```diff
@@ -85,6 +85,7 @@ class Wikia {
 
 	const COMMUNITY_WIKI_ID = 177; // community.wikia.com
 	const NEWSLETTER_WIKI_ID = 223496; // wikianewsletter.wikia.com
```
Comment (shaedrich) [inline includes/wikia/Wikia.php:87]: This wiki has one page only oO

Comment (macbre) [reply includes/wikia/Wikia.php:87]: But way more [images](http://wikianewsletter.wikia.com/wiki/Special:Images)

File: extensions/wikia/Search/WikiaSearchAjaxController.class.php [hunk 2]
```diff
@@ -27,10 +27,7 @@ public function getNextResults() {
 		$crossWikia = $this->request->getBool( 'crossWikia' );
 		$hub = false;
 
-		// TODO: refactor to avoid using wgEnableWikiaHomePageExt
-		$isCorporateWiki = !empty( $this->wg->EnableWikiaHomePageExt );
-
-		$isInterWiki = $crossWikia ? true : $isCorporateWiki;
+		$isInterWiki = $crossWikia ? true : WikiaPageType::isCorporatePage();
```
Comment (mszabo-wikia) [inline extensions/wikia/Search/WikiaSearchAjaxController.class.php]: `$crossWikia || WikiaPageType::isCorporatePage();` ?

File: extensions/wikia/Search/WikiaSearchController.class.php [hunk 3]
```diff
@@ -881,8 +881,7 @@ protected function templateExists( $name ) {
 	 * @see SearchControllerTest::testIsCorporateWiki
 	 */
 	protected function isCorporateWiki() {
-		// TODO: refactor to avoid using wgEnableWikiaHomePageExt
-		return !empty( $this->wg->EnableWikiaHomePageExt );
+		return WikiaPageType::isCorporatePage();
```
Comment (mszabo-wikia) [inline extensions/wikia/Search/WikiaSearchController.class.php:884]: we could kill this method wrapper here and use `WikiaPageType::isCorporatePage` directly

**Review by mszabo-wikia — CHANGES_REQUESTED** [review]

File: extensions/wikia/Search/WikiaSearchController.class.php (hunk 3 above)
Comment (macbre) [reply extensions/wikia/Search/WikiaSearchController.class.php:884]: I'm going to review all uses of `WikiaPageType::isCorporatePage` and eventually get rid of it. Search is the only place that has a different functionality on www.wikia.com.

File: includes/wikia/Wikia.php (hunk 1 above)
Comment (shaedrich) [reply includes/wikia/Wikia.php:87]: So it's just a image storage with no other use? They could be stored elsewhere.

**Pull request merged**